src/lib/operators/insert.cpp

```diff
@@ -168,6 +170,15 @@ std::shared_ptr<const Table> Insert::_on_execute(std::shared_ptr<TransactionCont
   }
   // TODO(all): make compress chunk thread-safe; if it gets called here by another thread, things will likely break.
 
+  if (transaction_context_is_set()) {
```

Comment:
Why do you need this if statement? `AbstractReadWriteOperator::execute` already has an Assert to check if the `transaction_context` is set.

Reply:
Makes sense, will be removed.